test/extended/builds/valuefrom.go

```diff
@@ -0,0 +1,112 @@
+	)
+
+	g.JustBeforeEach(func() {
+		g.By("waiting for builder service account")
+		err := exutil.WaitForBuilderAccount(oc.KubeClient().Core().ServiceAccounts(oc.Namespace()))
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("waiting for openshift namespace imagestreams")
+		err = exutil.WaitForOpenShiftNamespaceImageStreams(oc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("creating test image stream")
+		err = oc.Run("create").Args("-f", testImageStreamFixture).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("creating test secret")
+		err = oc.Run("create").Args("-f", secretFixture).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("creating test configmap")
+		err = oc.Run("create").Args("-f", configmapFixture).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+	})
+
+	g.It("should successfully process valueFrom in s2i build strategy environment variables", func() {
+
+		g.By("creating test successful build config")
+		err := oc.Run("create").Args("-f", successfulSTIBuildValueFrom).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("starting test build")
+		br, err := exutil.StartBuildAndWait(oc, "mys2itest")
+		o.Expect(err).NotTo(o.HaveOccurred())
+		br.AssertSuccess()
+
+		logs, _ := br.Logs()
+		o.Expect(logs).To(o.ContainSubstring("FIELDREF_ENV=mys2itest-1"))
+		o.Expect(logs).To(o.ContainSubstring("CONFIGMAPKEYREF_ENV=myvalue"))
+		o.Expect(logs).To(o.ContainSubstring("SECRETKEYREF_ENV=developer"))
+		o.Expect(logs).To(o.ContainSubstring("FIELDREF_CLONE_ENV=mys2itest-1"))
+		o.Expect(logs).To(o.ContainSubstring("FIELDREF_CLONE_CLONE_ENV=mys2itest-1"))
+		o.Expect(logs).To(o.ContainSubstring("UNAVAILABLE_ENV=$(SOME_OTHER_ENV"))
+		o.Expect(logs).To(o.ContainSubstring("ESCAPED_ENV=$(MY_ESCAPED_VALUE)"))
+
+	})
+
+	g.It("should successfully process valueFrom in docker build strategy environment variables", func() {
+
+		g.By("creating test successful build config")
+		err := oc.Run("create").Args("-f", successfulDockerBuildValueFrom).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("starting test build")
+		br, err := exutil.StartBuildAndWait(oc, "mydockertest")
+		o.Expect(err).NotTo(o.HaveOccurred())
+		br.AssertSuccess()
+
+		logs, _ := br.Logs()
+		o.Expect(logs).To(o.ContainSubstring("\"FIELDREF_ENV\" \"mydockertest-1\""))
+		o.Expect(logs).To(o.ContainSubstring("\"CONFIGMAPKEYREF_ENV\" \"myvalue\""))
+		o.Expect(logs).To(o.ContainSubstring("\"SECRETKEYREF_ENV\" \"developer\""))
+		o.Expect(logs).To(o.ContainSubstring("\"FIELDREF_CLONE_ENV\" \"mydockertest-1\""))
+		o.Expect(logs).To(o.ContainSubstring("\"FIELDREF_CLONE_CLONE_ENV\" \"mydockertest-1\""))
+		o.Expect(logs).To(o.ContainSubstring("\"UNAVAILABLE_ENV\" \"$(SOME_OTHER_ENV)\""))
+		o.Expect(logs).To(o.ContainSubstring("\"ESCAPED_ENV\" \"$(MY_ESCAPED_VALUE)\""))
+
+	})
+
+	g.It("should fail processing ResourceFieldRef in s2i build strategy environment variables", func() {
```

Comment:
yeah these tests (and the related resource files) should be removed.

What we do need a new test (probably integration or extended) for is the case where the EnvVar reference is valid, but can't be resolved successfully.  We need to make sure the build is failed and the failure message/reason is set correctly.